src/Dependencies/Collections/Extensions/IEnumerableExtensions.cs

```diff
@@ -866,13 +694,122 @@ internal static class Predicates<T>
 
 namespace System.Linq
 {
-    /// <summary>
-    /// Declare the following extension methods in System.Linq namespace to avoid accidental boxing of ImmutableArray{T} that implements IEnumerable{T}.
-    /// The boxing would occur if the methods were defined in Roslyn.Utilities and the file calling these methods has <c>using Roslyn.Utilities</c>
-    /// but not <c>using System.Linq</c>.
-    /// </summary>
-    internal static class EnumerableExtensions
+    /// <remarks>
+    /// Defines follyfil methods and overloads or alternative names of existing methods defined in System.Linq.
```

Comment:
*polyfill. (I can't make a suggestion from context of reviewing a commit, sorry.)